### PR TITLE
SSL clang-tidy fixes

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: cpp/src/Ice/SSL/SecureTransportEngine.cpp [hunk 1]
```diff
@@ -706,10 +707,6 @@ SecureTransport::SSLEngine::createServerAuthenticationOptions() const
     // It is safe to capture 'this' in the callbacks below as SSLEngine is managed by the communicator
     // and is guaranteed to outlive all connections.
     return ServerAuthenticationOptions{
```
Comment (pepone) [inline cpp/src/Ice/SSL/SecureTransportEngine.cpp:709]: These initializers were out of order.

File: cpp/src/Ice/SSL/OpenSSLTransceiverI.h [hunk 2]
```diff
@@ -37,17 +37,18 @@ namespace Ice::SSL::OpenSSL
...
+        [[nodiscard]] IceInternal::SocketOperation closing(bool, std::exception_ptr) final;
         void close() final;
-        IceInternal::SocketOperation write(IceInternal::Buffer&) final;
-        IceInternal::SocketOperation read(IceInternal::Buffer&) final;
-        std::string protocol() const final;
-        std::string toString() const final;
-        std::string toDetailedString() const final;
-        Ice::ConnectionInfoPtr getInfo(bool incoming, std::string adapterName, std::string connectionId) const final;
+        [[nodiscard]] IceInternal::SocketOperation write(IceInternal::Buffer&) final;
+        [[nodiscard]] IceInternal::SocketOperation read(IceInternal::Buffer&) final;
```
Comment (externl) [inline cpp/src/Ice/SSL/OpenSSLTransceiverI.h:46]: Why are these `nodiscard`? Inheritance?  Seems to be missing from `SecureTransportTranscieverI.h`. What's the pattern for overrides and adding `nodiscard`?

Comment (bernardnormier) [reply cpp/src/Ice/SSL/OpenSSLTransceiverI.h:46]: See https://stackoverflow.com/questions/49576298/are-function-attributes-inherited

However, it appears that clang-tidy treats `[[nodiscard]]` as inherited.

File: cpp/src/Ice/SSL/SchannelTransceiverI.h [hunk 3]
```diff
@@ -33,20 +33,21 @@ namespace Ice::SSL::Schannel
             const std::string&,
             const Ice::SSL::ClientAuthenticationOptions&);
         ~TransceiverI();
-        IceInternal::NativeInfoPtr getNativeInfo() final;
-        IceInternal::SocketOperation initialize(IceInternal::Buffer&, IceInternal::Buffer&) final;
-        IceInternal::SocketOperation closing(bool, std::exception_ptr) final;
+        [[nodiscard]] IceInternal::NativeInfoPtr getNativeInfo() final;
```
Comment (bernardnormier) [inline cpp/src/Ice/SSL/SchannelTransceiverI.h:36]: I assume these `[[nodiscard]]`` are not lint-related?

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**